profiler/upload.go

```diff
@@ -69,10 +70,18 @@ func (p *profiler) doRequest(bat batch) error {
 	if err != nil {
 		return err
 	}
+	ctx := context.Background()
+	if p.cfg.uploadTimeout != 0 {
+		var cancel func()
+		ctx, cancel = context.WithTimeout(context.Background(), p.cfg.uploadTimeout)
+		defer cancel()
+	}
+	// TODO(fg) use NewRequestWithContext once go 1.12 support is dropped.
 	req, err := http.NewRequest("POST", p.cfg.targetURL, body)
```

Comment:
We've noticed in dd-trace-go that sometimes users overwrite the `http.DefaultClient`, in our case with a traced one, and that'd create an inception of undesired traces. For this reason we've not used it, and allowed users to even set the client for the package themselves.

Reply:
We're not using `http.DefaultClient` here. See https://github.com/DataDog/dd-trace-go/blob/3db6059a8c1769f36f8d2a361ff7e8ffa3d60281/profiler/options.go#L59-L75

Reply:
Ah, my bad 🤦 Ignore me.